File: javatest/omnia/data/cache/MemoizedIntTest.java
package omnia.data.cache;

import static org.junit.Assert.assertTrue;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

@RunWith(JUnit4.class)
public class MemoizedIntTest {

  @Test
  public void memoize_didReturnSimpleIntMemoizer() {
    assertTrue(MemoizedInt.memoize(() -> 132) instanceof SimpleIntMemoizer);
  }
}

File: omnia/src/commonTest/omnia/data/cache/MemoizedIntTest.kt
package omnia.data.cache

import kotlin.test.Test
import omnia.util.test.fluent.Assertion.Companion.assertThat
import omnia.util.test.fluent.isA

class MemoizedIntTest {

  @Test
  fun memoize_didReturnSimpleIntMemoizer() {
    assertThat(MemoizedInt.memoize { 132 }).isA(SimpleIntMemoizer::class)
  }
}